src/datasets/iterable_dataset.py

```diff
@@ -78,6 +83,11 @@ def _shuffle_kwargs(rng: np.random.Generator, kwargs: dict) -> dict:
     return shuffled_kwargs
 
 
+def _shard_kwargs(shard_idx: int, kwargs: dict) -> dict:
+    """Return a copy of the input kwargs but with only one shard"""
+    return {key: [value[shard_idx]] if isinstance(value, list) else value for key, value in kwargs.items()}
```

Comment:
If I'm not mistaken, an index error can happen here in the case of different-sized lists in `gen_kwargs` (`n_shards` is equal to the max length of these lists), so we have to check if all the list sizes match, no?

Reply:
Maybe we should simply disallow users to use several lists of different sizes completely, to make things simple ? And make it clear in the docs that LISTS == SHARDS ?

I couldn't find a single dataset with lists of different sizes.

For now I'll just raise an error if lists are of different sizes in `_shard_kwargs`

Reply:
Yes, that sounds good!